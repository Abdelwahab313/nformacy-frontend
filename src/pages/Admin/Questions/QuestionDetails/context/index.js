import React, { createContext, useContext, useReducer } from 'react';
import QuestionActionTypes from './actionTypes';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      const updatedQuestion = { ...state.questionDetails, ...action.payload };
      return { ...state, questionDetails: updatedQuestion };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext, QuestionProvider };
