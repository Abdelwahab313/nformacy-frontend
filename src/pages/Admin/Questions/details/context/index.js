import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

export const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      return { ...state, questionDetails: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext as useAuth, QuestionProvider as AuthProvider };
