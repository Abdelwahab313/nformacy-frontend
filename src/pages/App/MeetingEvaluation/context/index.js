import React, { createContext, useContext, useReducer } from 'react';
// import CallEvaluationActionTypes from './actionTypes';

const defaultValues = {
  comment: '',
  ratingEvaluations: {}
};

const CallEvaluationContext = createContext();

const CallEvaluationProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = CallEvaluationContext;
  return (
    <Provider value={useReducer(CallEvaluationReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useCallEvaluationContext = () => {
  const context = useContext(CallEvaluationContext);
  if (context === undefined) {
    throw new Error('useCallEvaluationContext must be used within a CallEvaluationContext');
  }
  return context;
};

const CallEvaluationReducer = (state, action) => {
  switch (action.type) {

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useCallEvaluationContext, CallEvaluationProvider };
