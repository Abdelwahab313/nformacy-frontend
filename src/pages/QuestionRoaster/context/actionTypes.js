const QuestionRoasterActionTypes = {
  UPDATE_QUESTIONS: 'UPDATE_QUESTIONS',
  ADD_FIELD_FILTER: 'ADD_FIELD_FILTER',
  REMOVE_FIELD_FILTER: 'REMOVE_FIELD_FILTER',
  RESET_FIELDS_FILTERS: 'RESET_FIELDS_FILTERS',
  ADD_LANGUAGE_FILTER: 'ADD_LANGUAGE_FILTER',
};

export default QuestionRoasterActionTypes;
