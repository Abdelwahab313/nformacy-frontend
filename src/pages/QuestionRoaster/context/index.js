import React, { createContext, useContext, useReducer } from 'react';
import QuestionRoasterActionTypes from './actionTypes';

const defaultValues = {
  questions: [],
  fieldsFilters: [],
  languageFilter: '',
};

const QuestionRoasterContext = createContext();

const QuestionRoasterProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionRoasterContext;
  return (
    <Provider value={useReducer(QuestionRoasterReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionRoasterContext = () => {
  const context = useContext(QuestionRoasterContext);
  if (context === undefined) {
    throw new Error(
      'useQuestionRoasterContext must be used within a QuestionRoasterContext',
    );
  }
  return context;
};

const QuestionRoasterReducer = (state, action) => {
  switch (action.type) {
    case QuestionRoasterActionTypes.UPDATE_QUESTIONS:
      return { ...state, questions: [...action.payload] };
    case QuestionRoasterActionTypes.ADD_FIELD_FILTER:
      const newFilters = [...state.fieldsFilters];
      if (!state.fieldsFilters.includes(action.payload)) {
        newFilters.push(action.payload);
      }
      return {
        ...state,
        fieldsFilters: newFilters,
      };
    case QuestionRoasterActionTypes.REMOVE_FIELD_FILTER:
      const filtersAfterRemove = state.fieldsFilters.filter(
        (filter) => filter !== action.payload,
      );
      return { ...state, fieldsFilters: filtersAfterRemove };
    case QuestionRoasterActionTypes.RESET_FIELDS_FILTERS:
      return { ...state, fieldsFilters: [] };
    case QuestionRoasterActionTypes.ADD_LANGUAGE_FILTER:
      return { ...state, languageFilter: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionRoasterContext, QuestionRoasterProvider };
