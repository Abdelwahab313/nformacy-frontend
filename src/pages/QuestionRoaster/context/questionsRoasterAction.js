import QuestionRoasterActionTypes from './actionTypes';

export const updateFetchedQuestions = (dispatch, questions) => {
  dispatch({
    type: QuestionRoasterActionTypes.UPDATE_QUESTIONS,
    payload: questions,
  });
};

export const addFieldFilter = (dispatch, field) => {
  dispatch({
    type: QuestionRoasterActionTypes.ADD_FIELD_FILTER,
    payload: field,
  });
};

export const removeFieldFilter = (dispatch, field) => {
  dispatch({
    type: QuestionRoasterActionTypes.REMOVE_FIELD_FILTER,
    payload: field,
  });
};

export const addLanguageFilter = (dispatch, language) => {
  dispatch({
    type: QuestionRoasterActionTypes.ADD_LANGUAGE_FILTER,
    payload: language,
  });
};

export const resetFieldsFilters = (dispatch) => {
  dispatch({
    type: QuestionRoasterActionTypes.RESET_FIELDS_FILTERS,
  });
};
