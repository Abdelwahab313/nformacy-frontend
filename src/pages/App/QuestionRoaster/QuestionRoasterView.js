import React from 'react';
import { Grid } from '@material-ui/core';
import { useStyles } from 'styles/questionRoasterStyles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import useQuestionsFilter from './hooks/useQuestionsFilter';
import CircularProgress from '@material-ui/core/CircularProgress';
import QuestionView from './subComponents/QuestionView';
import QuestionsFilter from 'pages/App/QuestionRoaster/subComponents/QuestionsFilter';
import Typography from '@material-ui/core/Typography';
import { QuestionRoasterProvider } from './context';
import BreadcrumbsCustomSeparator from '../../../components/breadcrumbs/Breadcrumbs';
import useLocale from '../../../hooks/localization/useLocale';
import DIRECTION from '../../../constants/direction';
import { useTranslation } from 'react-i18next';

const QuestionRoasterView = () => {
  const { filteredQuestions, loading } = useQuestionsFilter();
  const { locale } = useLocale();
  const { t } = useTranslation();
  const classes = useStyles();

  if (loading) {
    return (
      <div id='progressBar' className={classes.progressContainer}>
        <CircularProgress />
      </div>
    );
  }

  return (
    <Grid id='questionRoasterMainContainer' container dir={DIRECTION[locale]}>
      <Grid container className={classes.bannerStyles} direction={'column'}>
        <Grid item id={'question-roaster-banner'}>
          <Typography
            id={'question-roaster-header'}
            className={classes.bannerFontStyles}>
            {t('questionRoaster:title')}
          </Typography>
        </Grid>
        <Grid item className={classes.searchGridStyles}>
          <Paper
            component='form'
            className={classes.searchBarContainer}
            id={'question-roaster-search-bar'}>
            <InputBase
              className={classes.searchInput}
              placeholder={t('common:enterKeyword')}
              inputProps={{
                'aria-label': t('common:searchKeyword'),
                id: 'search',
              }}
            />
            <IconButton
              type='submit'
              className={classes.iconButton}
              aria-label={t('common:search')}>
              <SearchIcon color={'secondary'} />
            </IconButton>
          </Paper>
        </Grid>
      </Grid>
      <Grid container justify={'center'}>
        <Grid item xs={10}>
          <BreadcrumbsCustomSeparator pageName={t('questionRoaster:title')} />
        </Grid>
      </Grid>
      <Grid
        container
        justify={'center'}
        id={'questions-roaster-filters-container'}>
        <QuestionsFilter />
        <Grid item xs={12} sm={10}>
          <Grid container>
            {filteredQuestions?.map((question, key) => {
              return (
                <QuestionView
                  questionDetails={question}
                  key={key}
                  isSubmitVisible={true}
                />
              );
            })}
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

const QuestionRoaster = (props) => {
  return (
    <QuestionRoasterProvider>
      <QuestionRoasterView {...props} />
    </QuestionRoasterProvider>
  );
};
export default QuestionRoaster;
