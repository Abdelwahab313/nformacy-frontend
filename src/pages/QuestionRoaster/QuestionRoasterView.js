import React from 'react';
import { Grid } from '@material-ui/core';
import { useStyles } from 'styles/questionRoasterStyles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import useQuestionsFilter from './hooks/useQuestionsFilter';
import CircularProgress from '@material-ui/core/CircularProgress';
import QuestionView from './subComponents/QuestionView';
import QuestionsFilter from 'pages/QuestionRoaster/subComponents/QuestionsFilter';
import Typography from '@material-ui/core/Typography';
import t from '../../locales/en/questionRoaster';
import { QuestionRoasterProvider } from './context';
import BreadcrumbsCustomSeparator from '../../components/breadcrumbs/Breadcrumbs';
import useLocale from '../../hooks/localization/useLocale';
import DIRECTION from '../../constants/direction';

const QuestionRoasterView = () => {
  const { filteredQuestions, loading } = useQuestionsFilter();
  const { locale } = useLocale();
  const classes = useStyles();

  if (loading) {
    return (
      <div id='progressBar' className={classes.progressContainer}>
        <CircularProgress/>
      </div>
    );
  }

  return (
    <Grid id='questionRoasterMainContainer' container dir={DIRECTION[locale]}>
      <Grid item xs={12} className={classes.bannerStyles} direction={'column'}>
        <Grid item id={'question-roaster-banner'}>
          <Typography
            id={'question-roaster-header'}
            className={classes.bannerFontStyles}>
            {t['questionRoaster']}
          </Typography>
        </Grid>
        <Grid item className={classes.searchGridStyles}>
          <Paper
            component='form'
            className={classes.searchBarContainer}
            id={'question-roaster-search-bar'}>
            <InputBase
              className={classes.searchInput}
              placeholder={t['enterKeyword']}
              inputProps={{ 'aria-label': 'search by keyword', id: 'search' }}
            />
            <IconButton
              type='submit'
              className={classes.iconButton}
              aria-label='search'>
              <SearchIcon color={'secondary'}/>
            </IconButton>
          </Paper>
        </Grid>
      </Grid>
      <Grid container justify={'center'}>
        <Grid item xs={10}>
          <BreadcrumbsCustomSeparator pageName={'Question Roaster'}/>
        </Grid>
      </Grid>
      <Grid
        container
        justify={'center'}
        id={'questions-roaster-filters-container'}>
        <QuestionsFilter />
        <Grid item xs={12} sm={10}>
          <Grid container>
            {filteredQuestions?.map((question, key) => {
              return (
                <QuestionView
                  questionDetails={question}
                  key={key}
                  isSubmitVisible={true}
                />
              );
            })}
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

const QuestionRoaster = (props) => {
  return (
    <QuestionRoasterProvider>
      <QuestionRoasterView {...props} />
    </QuestionRoasterProvider>
  );
};
export default QuestionRoaster;
