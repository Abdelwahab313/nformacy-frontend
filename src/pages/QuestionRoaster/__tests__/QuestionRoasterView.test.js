import React from 'react';
import { render, waitFor } from '@testing-library/react';
import QuestionRoasterView from '../QuestionRoasterView';
import { fetchOpenedQuestions } from 'apis/questionsAPI';
import { AuthProvider } from '../../auth/context/auth';
import { LocaleProvider } from '../../../hooks/localization/context';

jest.mock('react-i18next', () => {
  return {
    ...require.requireActual('react-i18next'),
    useTranslation: jest.fn().mockReturnValue({
      i18n: {
        getFixedT: () => (() => {
        }),
      },
    }),
  };
});

jest.mock('apis/questionsAPI', () => ({
  __esModule: true, // this property makes it work
  default: 'mockedDefaultExport',
  fetchOpenedQuestions: jest.fn().mockResolvedValue({
    data: [
      {
        assignedAdviserId: 2,
        assignmentType: 'project',
        content:
          'From the viewpoint of what you can do, therefore, languages do differ - but the differences are limited. For example, Python and Ruby provide almost the same power to the programmer.',
        createdAt: '2020-09-14T09:25:55.689Z',
        currentActionTime: '2020-09-14T22:02:28.411Z',
        field: [{ value: 'marketingAndPR', label: 'Marketing and PR' }],
        hoursToCloseAnswers: null,
        hoursToReviewAndEdit: null,
        id: 1,
        industry: { value: 'consulting', label: 'Consulting' },
        isApproved: true,
        mediaId: null,
        referenceNumber: 2000100,
        state: 'pending_adviser_acceptance',
        subfield: [{ value: 'marketResearch', label: 'Market Research' }],
        title: 'Elmer',
      },
    ],
  }),
}));

describe('Question Roaster View', () => {
  it('should match snapshot', async () => {
    const { asFragment } = render(
      <AuthProvider>
        <LocaleProvider initialLocale={'en'}>
          <QuestionRoasterView/>
        </LocaleProvider>
      </AuthProvider>)
      ;
    await waitFor(() => expect(fetchOpenedQuestions).toHaveBeenCalledTimes(1));
    expect(asFragment()).toMatchSnapshot();
  });
});
