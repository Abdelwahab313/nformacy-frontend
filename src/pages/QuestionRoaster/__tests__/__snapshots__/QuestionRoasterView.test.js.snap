// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Roaster View should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
    dir="ltr"
    id="questionRoasterMainContainer"
  >
    <div
      class="MuiGrid-root bannerStyles MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
        id="question-roaster-banner"
      >
        <p
          class="MuiTypography-root bannerFontStyles MuiTypography-body1"
          id="question-roaster-header"
        >
          Question Roaster
        </p>
      </div>
      <div
        class="MuiGrid-root searchGridStyles MuiGrid-item"
      >
        <form
          class="MuiPaper-root searchBarContainer MuiPaper-elevation1 MuiPaper-rounded"
          id="question-roaster-search-bar"
        >
          <div
            class="MuiInputBase-root searchInput"
          >
            <input
              aria-label="search by keyword"
              class="MuiInputBase-input"
              id="search"
              placeholder="Enter Keyword"
              type="text"
              value=""
            />
          </div>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiIconButton-root iconButton"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      >
        <div
          class="root"
        >
          <nav
            aria-label="breadcrumb"
            class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            id="question-roaster-breadcrumbs"
          >
            <ol
              class="MuiBreadcrumbs-ol"
            >
              <li
                class="MuiBreadcrumbs-li"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  href="/"
                >
                  <p
                    class="MuiTypography-root inactiveBreadcrumb MuiTypography-body1"
                    id="home-breadcrumb"
                  >
                    Home
                  </p>
                </a>
              </li>
              <li
                aria-hidden="true"
                class="MuiBreadcrumbs-separator"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </li>
              <li
                class="MuiBreadcrumbs-li"
              >
                <p
                  class="MuiTypography-root activeBreadcrumb MuiTypography-body1 MuiTypography-colorPrimary"
                >
                  Question Roaster
                </p>
              </li>
            </ol>
          </nav>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      id="questions-roaster-filters-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10"
      >
        <div
          class="MuiGrid-root questionsCategoriesContainerDesktop MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
          id="filters"
        >
          <div
            class="MuiGrid-root fieldsFiltersContainer MuiGrid-item MuiGrid-grid-md-10"
          >
            <div
              class="activeFilterStyle"
            >
              All
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-0"
            >
              Finance
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-1"
            >
              Human Resource
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-2"
            >
              Marketing and PR
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-3"
            >
              Formal Education
            </div>
            <div>
              <button
                aria-controls="long-menu"
                aria-haspopup="true"
                aria-label="more"
                class="MuiButtonBase-root MuiIconButton-root threeDotButton"
                id="more-options-menu"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    id="more-options-icon"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root languageFilterContainer MuiGrid-item MuiGrid-grid-md-2"
          >
            <div>
              <button
                aria-controls="simple-menu"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text dropdownDesktop"
                id="question-language-filter"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Choose Langauge
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    id="expand-menu-icon"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root filterWrapper MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root questionsCategoriesContainerMobile MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-9"
            id="filters-mobile"
          >
            <div
              class="activeFilterStyle"
            >
              All
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-0"
            >
              Finance
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-1"
            >
              Human Resource
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-2"
            >
              Marketing and PR
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-3"
            >
              Formal Education
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-4"
            >
              Development
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-5"
            >
              Entrepreneurship & Start ups
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-6"
            >
              Strategy
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-7"
            >
              Operations Management
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-8"
            >
              Supply Chain
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-9"
            >
              Digitization
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-10"
            >
              Sales and Customer Care
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-11"
            >
              Legal
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-12"
            >
              FinTech
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-13"
            >
              Environment
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-14"
            >
              Project Management
            </div>
            <div
              class="inactiveFilterStyle"
              id="filters-15"
            >
              Analytics
            </div>
          </div>
          <div
            class="MuiGrid-root nextIconSlider MuiGrid-item MuiGrid-grid-xs-1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root nextIconSize MuiSvgIcon-colorPrimary"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
          <div
            class="MuiGrid-root dropdownMobile MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-1"
          >
            <div>
              <button
                aria-controls="simple-menu"
                aria-haspopup="true"
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                id="question-language-filter-mobile"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  E
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root mainContainer MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
            id="questionRoasterMainContainer"
          >
            <div
              class="MuiGrid-root questionContainer MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
              >
                <div
                  class="MuiGrid-root questionTextWrapper MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-2"
                  >
                    <p
                      class="MuiTypography-root referenceNumberStyle MuiTypography-body1"
                      id="question-2000100-referenceNumber"
                    >
                      # 2000100
                    </p>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-7"
                  >
                    <p
                      class="MuiTypography-root postDateStyle MuiTypography-body1"
                      id="question-2000100-postDate"
                    >
                      Sep 14
                    </p>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-6 MuiGrid-grid-md-9"
                  >
                    <p
                      class="MuiTypography-root questionTitle MuiTypography-body1"
                      id="question-2000100-title"
                    >
                      Elmer
                    </p>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3"
                    id="question-2000100-currentActionTime"
                  >
                    <div
                      class="MuiGrid-root timeContainer MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
                      >
                        <p
                          class="MuiTypography-root closedQuestion MuiTypography-body1"
                          id="question-2000100-closeIn"
                        />
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
                      >
                        <p
                          class="MuiTypography-root closedQuestion MuiTypography-body1"
                        >
                          Closed
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root flexContainer MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
                  >
                    <div
                      class="MuiGrid-root fieldContainer"
                    >
                      <div
                        class="MuiGrid-root"
                      >
                        <div
                          class="MuiChip-root sc-bdnylx bpbFSf fieldChip"
                          index="0"
                        >
                          <span
                            class="MuiChip-label"
                          >
                            <p
                              class="MuiTypography-root fieldChipText MuiTypography-body1"
                              id="questionMajorFields-2000100-0"
                            >
                              Marketing and PR
                            </p>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
                  >
                    <div
                      class="MuiGrid-root questionContentField"
                      id="question-2000100-content"
                    >
                      <div>
                        <span
                          width="0"
                        >
                          <span />
                          <span>
                            <div>
                              No Content
                            </div>
                          </span>
                          <span
                            style="position: fixed; visibility: hidden; top: 0px; left: 0px;"
                          >
                            <span>
                              ... 
                              <a
                                class="anchor"
                                href="#"
                              >
                                <div
                                  class="MuiGrid-root icons MuiGrid-container"
                                >
                                  <div>
                                     Show More 
                                  </div>
                                  <svg
                                    aria-hidden="true"
                                    class="MuiSvgIcon-root icon"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                                    />
                                  </svg>
                                </div>
                              </a>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root questionAssignmentTypeContainer MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-md-10"
                    id="question-2000100-assignment"
                  >
                    <div
                      class="MuiGrid-root assignmentTypeIcon"
                    >
                      <img
                        color="primary"
                        height="100%"
                        id="question-2000100-project"
                        src="[object Object]"
                        width="100%"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root answerButtonContainer MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-md-2"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained submitButton"
                      id="question-2000100-submit"
                      style="background-color: rgb(18, 87, 115); color: rgb(255, 255, 255);"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      />
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
